### ISSUE TITLE
envoy_reloadable_features_quic_send_server_preferred_address_to_all_clients deprecation

### ISSUE THREAD

**Issue by alyssawilk** [envoyproxy/envoy#36607]
Your change #32130 (quic: enable sending server preferred address to non-quiche clients) introduced a runtime guarded feature. It has been 6 months since the new code has been exercised by default, so it's time to remove the old code path. This issue tracks source code cleanup so we don't forget.

**Comment by ggreenway** [comment]
Code comment on this flag: 

```
// Ignore the automated "remove this flag" issue: we should keep this for 1 year. Confirm with
// @danzh2010 or @RyanTheOptimist before removing.
```

Original commit date: 2024-02-12. Deferring until then.

**Comment by alyssawilk** [comment]
cc @danzh2010 @RyanTheOptimist just to confirm, we want to keep this knob around?

**Comment by RyanTheOptimist** [comment]
> Yes, our internal Envoy server turns it off explicitly and still relies the feature on an connection option.

Do we have plans to turn this on internally? If not, should we land a "real" API for this instead of a reloadable feature?

**Comment by danzh2010** [comment]
Yes, definitely we plan to enable SPA for all on server side as well, no need for a real API as the connection option will be deprecated one day.

**Comment by ggreenway** [comment]
@danzh2010 should I assign this one to you, and whenever you don't need the runtime flag anymore you can remove?

**Comment by danzh2010** [comment]
Yes, I can take ownership of this RUNTIME.
